TypeSystem/flow-analysis/promotion_switch_A04_t02.dart

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion For now (May, 2025) promotion in switch statement and
+/// expressions are not specified yet.
+/// TODO (sgrekhov): update when specified
+///
+/// @description Checks that pattern match adds type of interest to the switch
+/// statement/expression variable.
+/// @author sgrekhov22@gmail.com
+
+import '../../Utils/static_type_helper.dart';
+
+test1(Object? x) {
+  switch (x) {
+    case int _:
+      print("");
+    case _:
+      x = 42;
```

Comment:
At this point, `x` has version `x#2` because of the assignment, and we have

> Since the version of `e` is no longer `o#1`, `o` is not promoted.

But that would be the type tests implied by pattern matching (because they are still applied to the value of `x#1`, so we can't allow a pattern match on `int _` to say anything about the new value of `x`).

It should still be the case that `int` is a type of interest for `x`, and `x` should still be promoted via assignment of an expression whose static type is of interest (or it has a unique best supertype that is). So line 20 _should_ promote `x` to `int`.

OK!